src/pages/studyView/tabs/SummaryTab.tsx

```diff
@@ -204,6 +213,20 @@ export class StudySummaryTab extends React.Component<
                     props.promise = this.store.getGenericAssayChartDataCount(
                         chartMeta
                     );
+                } else if (
```

Comment:
weird that gene specific chart should be exclusive with, say "generic assay chart".    Perhaps it's just semantics but we should improve the name then.

Reply:
how much effort would it take to refactor/break up this switch statement (which then has the internal if else statement.  it's just way too long and complicated.